animations/lib/src/misc/carousel.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+/// This sample is based on the `coverflow` package
+/// https://github.com/efortuna/coverflow
+class CarouselDemo extends StatelessWidget {
+  static String routeName = '/misc/carousel';
+
+  static List<String> fileNames = [
+    'assets/eat_cape_town_sm.jpg',
+    'assets/eat_new_orleans_sm.jpg',
+    'assets/eat_sydney_sm.jpg',
+  ];
+
+  final List<Widget> data = [
+    ...fileNames.map((file) => Image.asset(file, fit: BoxFit.cover))
+  ];
+
+  @override
+  Widget build(context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Carousel Demo'),
+      ),
+      body: Carousel(itemBuilder: widgetBuilder),
+    );
+  }
+
+  Widget widgetBuilder(context, int index) {
+    return data[index % data.length];
+  }
+}
+
+typedef void OnCurrentItemChangedCallback(int currentItem);
+
+class Carousel extends StatefulWidget {
+  final IndexedWidgetBuilder itemBuilder;
+
+  const Carousel({Key key, @required this.itemBuilder});
+
+  @override
+  _CarouselState createState() => _CarouselState();
+}
+
+class _CarouselState extends State<Carousel> {
+  final _height = 525;
+  final _width = 700;
+  PageController _controller;
+  int _currentPage;
+  bool _pageHasChanged = false;
+
+  @override
+  initState() {
+    super.initState();
+    _currentPage = 0;
+    _controller = PageController(
+      viewportFraction: .85,
+      initialPage: _currentPage,
+    );
+  }
+
+  @override
+  Widget build(context) {
+    return PageView.builder(
+        onPageChanged: (value) {
+          setState(() {
+            _pageHasChanged = true;
+            _currentPage = value;
+          });
+        },
+        controller: _controller,
+        itemBuilder: (context, index) => builder(index));
```

Comment:
nit: Throw a comma in between the closing parens on this line to make flutter format indent better.